src/pages/library/dataBrowser-utils.js

```diff
@@ -123,14 +132,34 @@ export const useDataCatalog = () => {
   return { dataCatalog, refresh, loading }
 }
 
-export const importDataToWorkspace = datasets => {
-  // TODO (DC-284): Call data catalog to figure out what the format should be for importing to workspace
-  const format = 'snapshot'
-  Nav.history.push({
-    pathname: Nav.getPath('import-data'),
-    search: qs.stringify({
-      url: getConfig().dataRepoUrlRoot, format, referrer: 'data-catalog',
-      snapshotIds: _.map('dct:identifier', datasets)
-    })
-  })
+export const importDataToWorkspace = async (dataset, asyncHandler) => {
+  const routeOptions = await Utils.cond(
+    [isWorkspace(dataset), () => {
+      return {
+        pathname: Nav.getPath('import-data'),
+        search: qs.stringify({
+          format: 'catalog',
+          snapshotName: dataset['dct:title'],
+          catalogDatasetId: dataset.id
+        })
+      }
+    }],
+    [
+      isDatarepoSnapshot(dataset), async () => {
+        asyncHandler()
+        const jobInfo = await Ajax().DataRepo.snapshot(dataset['dct:identifier']).exportSnapshot()
+        await poll(async () => await Ajax().DataRepo.job(jobInfo.id).details(), 1000, jobStatus => jobStatus['job_status'] !== 'running')
```

Comment:
So, I was mostly following the mocks that Lou has laid out. I agree that it is a while to wait with such minimal feedback. If the user navigates away, it will actually pull them back to the `import-data` page when finished, regardless of what they're doing. This is probably bad UX, and it should probably cancel when the user leaves the page. I'll see what I can do for that.

We may want to change it, but I am inclined to get user feedback first, and go from there. We, sadly,

Reply:
Okay, here's the update from talking with Lou:
Before navigating, the user is going to need to click again (so there is a secondary click when it is ready)
We're also going to open a toast, that will say "your data is being prepared for analysis", in case the user clicks away. They can also click a link in the toast when the export is finished to go to the import page. This toast is going to be non-closable.